api_app/analyzers_manager/file_analyzers/virushee.py

```diff
@@ -0,0 +1,93 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import logging
+import time
+
+import requests
+
+from api_app.analyzers_manager.classes import FileAnalyzer
+from api_app.exceptions import AnalyzerRunException
+from tests.mock_utils import MockResponse, if_mock_connections, patch
+
+logger = logging.getLogger(__name__)
+
+
+class Virushee(FileAnalyzer):
+    """Analyze a file against Virushee API"""
+
+    def set_params(self, params):
+        self.session = requests.Session()
+        self.max_tries = 30
+        self.poll_distance = 10
+        self.request_url = "https://api.virushee.com/"
+
+    def run(self):
+        binary = self.read_file_bytes()
+        if not binary:
+            raise AnalyzerRunException("File is empty")
+        hash_exist = self.is_hash_avaliable()
+        task_id = self.__upload_file(binary)
+        result = self.__poll_status(task_id)
+        return result
+
+    def is_hash_avaliable(self) -> bool:
+        api_endpoint = f"file/hash/{self.md5}"
+        api_url = self.request_url + api_endpoint
+        response = self.session.get(api_url)
+        if response.status_code == 200:
+            return True
+        return False
+
+    def __upload_file(self, binary) -> str:
+        name_to_send = self.filename if self.filename else self.md5
+        files = {"file": (name_to_send, binary)}
+        upload_url = self.request_url + "file/upload"
+        response = self.session.post(upload_url, files=files)
+        if response.status_code == 422:
+            raise AnalyzerRunException(
+                "Virushee API returned an error: " + response.text
+            )
+        elif response.status_code == 200:
+            logger.info("HASH FOUND")
+            raise AnalyzerRunException(
+                "Virushee API returned an error: " + str(response.status_code)
```

Comment:
instead of manually checking for each status code, you should wrap the request in a try/except and use `response.raise_for_status()`. Like [this](https://github.com/intelowlproject/IntelOwl/blob/master/api_app/analyzers_manager/observable_analyzers/shodan.py#L33-L37).

Reply:
Thanks, for the improvement :)